src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoader.cs

```diff
@@ -377,13 +517,18 @@ public Bindings(ModelLoadContext ctx, DatabaseLoader parent)
                 //   byte: bool of whether this is a key type
                 //   for a key type:
                 //     ulong: count for key range
-                //   byte: bool of whether the source index is valid
-                //   for a valid source index:
-                //     int: source index
+                //   int: number of segments
+                //   foreach segment:
+                //     int: min
+                //     int: lim
+                //     byte: force vector (verWrittenCur: verIsVectorSupported)
                 int cinfo = ctx.Reader.ReadInt32();
                 Contracts.CheckDecode(cinfo > 0);
                 Infos = new ColInfo[cinfo];
 
+                // This dictionary is used only for detecting duplicated column names specified by user.
+                var nameToInfoIndex = new Dictionary<string, int>(Infos.Length);
```

Comment:
It doesn't look like this dictionary is used to detect duplicates. It just gets the names set on it, no one checks if there is a dupe.